server/server.go

```diff
@@ -622,8 +648,12 @@ func (s *Server) SetNamespaceConfig(name string, cfg NamespaceConfig) error {
 func (s *Server) DeleteNamespaceConfig(name string) error {
 	if n, ok := s.scheduleOpt.ns[name]; ok {
 		cfg := n.load()
-		delete(s.scheduleOpt.ns, name)
+		s.scheduleOpt.ns[name].store(&NamespaceConfig{})
```

Comment:
Delete is not atomic, It may cause `DATA RACE` error.
I have met it in running test.

Reply:
But this way may get error config? all default is zero.

Reply:
https://github.com/pingcap/pd/blob/9ba0e4453c3ae4581f2ea0d52c2f7803d57905ca/server/server.go#L578-L582

If `ns[name]` doesnt exist,it will return a `&NamespaceConfig{} `, so they are equivalent?

Reply:
got it.